### PR TITLE
[Improve] Unified Mapper

### PR DESCRIPTION
**Pull request by wuchunfu**
## What changes were proposed in this pull request

Issue Number: #1771  

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

### CODE REVIEW

**Comment by wuchunfu** [comment]
@wolfboys PTAL, thanks

**Comment by wolfboys** [comment]
> @wolfboys PTAL, thanks

Thanks for your hard work, I will review it later

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java [hunk 1]
```diff
@@ -1029,7 +1033,13 @@ public String getMain(Application application) {
     @Override
     @RefreshCache
     public boolean mapping(Application appParam) {
-        boolean mapping = this.baseMapper.mapping(appParam);
+        appParam.setAppId(appParam.getAppId());
+        appParam.setJobId(appParam.getJobId());
+        appParam.setState(14);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java]: you can improve the code: `appParam.setState(FlinkAppState.MAPPING.getValue());`

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java [hunk 2]
```diff
@@ -1029,7 +1033,13 @@ public String getMain(Application application) {
     @Override
     @RefreshCache
     public boolean mapping(Application appParam) {
-        boolean mapping = this.baseMapper.mapping(appParam);
+        appParam.setAppId(appParam.getAppId());
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java]: oh no, This code may have bugs, I recommend query sql write to mapper.xml

**Review by wolfboys — COMMENTED** [review]
review WIP

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java [hunk 3]
```diff
@@ -60,7 +63,10 @@ public boolean update(Setting setting) {
         try {
             String value = StringUtils.trimToNull(setting.getSettingValue());
             setting.setSettingValue(value);
-            this.baseMapper.updateByKey(setting);
+
+            LambdaQueryWrapper<Setting> queryWrapper = new LambdaQueryWrapper<>();
+            queryWrapper.eq(Setting::getSettingKey, setting.getSettingKey());
+            this.update(setting, queryWrapper);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java]: I'm not sure if other fields other than SettingValue are also updated

File: streampark-console/streampark-console-service/src/main/resources/mapper/core/ApplicationConfigMapper.xml [hunk 4]
```diff
@@ -29,4 +29,19 @@
...
+        where app_id = #{appId}
+    </select>
+
+    <select id="getEffective" resultType="org.apache.streampark.console.core.entity.ApplicationConfig" parameterType="java.lang.Long">
+        select s.*
+        from t_flink_config s
+        inner join t_flink_effective e
+        on s.id = e.target_id
+        where e.app_id = #{appId}
+            and e.target_type = 1
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/resources/mapper/core/ApplicationConfigMapper.xml]: need Improve the format

**Comment by wuchunfu** [comment]
@wolfboys PTAL, thanks

**Comment by wolfboys** [comment]
> @wolfboys PTAL, thanks

Thanks for your hard work, I will review it as soon as possible

**Review by wolfboys — COMMENTED** [review]
ApplicationConfig pass.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ProjectServiceImpl.java [hunk 5]
```diff
@@ -227,9 +234,37 @@ public boolean checkExists(Project project) {
...
+        LambdaQueryWrapper<Project> queryWrapper = new LambdaQueryWrapper<Project>()
             .eq(Project::getName, project.getName());
-        return this.baseMapper.selectCount(wrapper) > 0;
+        return this.baseMapper.selectCount(queryWrapper) > 0;
+    }
+
+    @Override
+    public void updateFailureBuildById(Project project) {
+        Project entity = new Project();
+        entity.setBuildState(2);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ProjectServiceImpl.java]: you can replace "2" to `BuildState.FAILED.getValue()`

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ProjectServiceImpl.java [hunk 6]
```diff
@@ -227,9 +234,37 @@ public boolean checkExists(Project project) {
...
+        LambdaQueryWrapper<Project> queryWrapper = new LambdaQueryWrapper<Project>()
+            .eq(Project::getId, project.getId());
+        this.update(entity, queryWrapper);
+    }
+
+    @Override
+    public void updateSuccessBuildById(Project project) {
+        Project entity = new Project();
+        entity.setLastBuild(new Date());
+        entity.setBuildState(1);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ProjectServiceImpl.java]: you can replace "1" to `BuildState.SUCCESSFUL.getValue()`

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/ProjectService.java [hunk 7]
```diff
@@ -56,4 +56,10 @@ public interface ProjectService extends IService<Project> {
 
     boolean checkExists(Project project);
 
+    void updateFailureBuildById(Project project);
+
+    void updateSuccessBuildById(Project project);
+
+    void updateStartBuildById(Project project);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/ProjectService.java]: Three methods can be simplified into one method:

```
void updateBuildById(Long projectId, BuildState state);
```
I recommend to put the updated implementation in `ProjectMapper.xml`

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java [hunk 8]
```diff
@@ -60,7 +63,12 @@ public boolean update(Setting setting) {
         try {
             String value = StringUtils.trimToNull(setting.getSettingValue());
             setting.setSettingValue(value);
-            this.baseMapper.updateByKey(setting);
+
+            Setting entity = new Setting();
+            entity.setSettingValue(setting.getSettingValue());
+            LambdaQueryWrapper<Setting> queryWrapper = new LambdaQueryWrapper<Setting>()
+                .eq(Setting::getSettingKey, setting.getSettingKey());
+            this.update(entity, queryWrapper);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java:71]: I recommend to move the implementation in mapper.xml

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java [hunk 9]
```diff
@@ -44,7 +45,9 @@ public class SettingServiceImpl extends ServiceImpl<SettingMapper, Setting>
 
     @Override
     public Setting get(String key) {
-        return baseMapper.get(key);
+        LambdaQueryWrapper<Setting> queryWrapper = new LambdaQueryWrapper<Setting>()
+            .eq(Setting::getSettingKey, key);
+        return this.getOne(queryWrapper);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java:50]: I recommend to move the implementation in mapper.xml

**Comment by wuchunfu** [comment]
@wolfboys PTAL, thanks

**Review by wolfboys — APPROVED** [review]
LGTM, thanks for your hard work

**Pull request merged**